server/memphis_handlers_monitoring.go

```diff
@@ -277,3 +281,157 @@ func (mh MonitoringHandler) GetStationOverviewData(c *gin.Context) {
 
 	c.IndentedJSON(200, response)
 }
+
+func (mh MonitoringHandler) GetSystemLogs(c *gin.Context) {
+	const amount = 100
+	const timeout = 3 * time.Second
+
+	var request models.SystemLogsRequest
+	ok := utils.Validate(c, &request, false, nil)
+	if !ok {
+		return
+	}
+
+	startSeq := uint64(request.StartIdx)
+	getLast := false
+	if request.StartIdx == -1 {
+		getLast = true
+	}
+
+	filterSubject, filterSubjectSuffix := _EMPTY_, _EMPTY_
+	switch request.LogType {
+	case "err":
+		filterSubjectSuffix = syslogsErrSubject
+	case "wrn":
+		filterSubjectSuffix = syslogsWarnSubject
+	case "inf":
+		filterSubjectSuffix = syslogsInfoSubject
+	}
+
+	if filterSubjectSuffix != _EMPTY_ {
+		filterSubject = syslogsStreamName + "." + filterSubjectSuffix
+	}
+
+	response, err := mh.S.GetSystemLogs(amount, timeout, getLast, startSeq, filterSubject)
+	if err != nil {
+		serv.Errorf("GetSystemLogs error: " + err.Error())
+		c.AbortWithStatusJSON(500, gin.H{"message": "Server error"})
+		return
+	}
+
+	c.IndentedJSON(200, response)
+}
+
+func min(x, y uint64) uint64 {
+	if x < y {
+		return x
+	}
+	return y
+}
+
+func (s *Server) GetSystemLogs(amount uint64,
+	timeout time.Duration,
+	fromLast bool,
+	startSeq uint64,
+	filterSubject string) (models.SystemLogsResponse, error) {
+	uid := s.memphis.nuid.Next()
+	durableName := "$memphis_fetch_logs_consumer_" + uid
+
+	if fromLast {
+		streamInfo, err := s.memphisStreamInfo(syslogsStreamName)
+		if err != nil {
+			return models.SystemLogsResponse{}, err
+		}
+		startSeq = min(streamInfo.State.Msgs-amount, uint64(1))
```

Comment:
not good, start seq should be the lastSEQ - amount. and note that that can be negative